### PR TITLE
[misc] goodbye grunt

### PR DESCRIPTION
**Pull request by das7pad**
## Description

For https://github.com/overleaf/issues/issues/3228

This PR is dropping all the grunt stuff used for
- installing services in combination with some shell scripts
- checking db access on instance boot
- creating/deleting users (broken since mongojs changes)
- updating (I've dropped this, we do not support in-place / in-docker upgrades)

To ease the migration, I've added thin wrapper script in the `$PATH` that shims the common grunt tasks and `grunt user:create-admin --email=foo@bar` still works.  

Moving forward we want to test all the tasks, to ensure that e.g. the user-admin stuff does not break without us noticing. The tasks have been migrated to web-internal from where they will be mirrored/pulled into server-ce/pro.

We need to interface with the service details from the `service.js` file (name/github-url/version/branch). I decided not to write yet another mixture of shell-scripts that query this file to then compute arguments for git/npm/etc. in bash. Instead, I am generating all the contents of these shell-script in NodeJS and pipe them to bash. That seems to be simpler than forking from NodeJS and handling all exit-codes, stdout/stderr handling/logging etc.

Note: Server-ce is currently broken at run-time due to bunyan quirks with some google-library (https://github.com/overleaf/real-time/pull/222). I've triggered all the builds to upgrade the services, PRs are coming tomorrow.
Bonus: We can close ALL the dependabot PRs.

## Related issues / Pull Requests

For https://github.com/overleaf/issues/issues/3228

### Manual testing performed

- put CI image in docker-compose file
- bind mount server-ce scripts from web: `dev-env/web/scripts/server-ce:/var/www/sharelatex/web/scripts/server-ce`
- start server ce
- editor works
- compiles work
- chat works

### CODE REVIEW

**Review by mserranom — APPROVED** [review]
Looks good, thanks. Pretty clever the `genScript.js` solution.

For a `3.0.0` release I think ditch the execution of this tasks from the container, and move everything to the toolkit (which is something we could do now Cc @ShaneKilkelly )

File: genScript.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const services = require('./services')
+
+console.log('#!/bin/bash')
+console.log('set -ex')
+
+switch (process.argv.pop()) {
+  case 'checkout':
+    for (const service of services) {
+      console.log(`git clone ${service.repo} ${service.name}`)
```
Comment (ShaneKilkelly) [inline genScript.js:9]: I'd be a bit worried about shell injection here, _but_, I doubt we'll ever have data that's not under our control, so...

File: genScript.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const services = require('./services')
+
+console.log('#!/bin/bash')
+console.log('set -ex')
```
Comment (ShaneKilkelly) [inline genScript.js:4]: Is there a reason for this script to be node generating bash, rather than just a bash script?

File: bin/grunt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Thin wrapper on old grunt tasks to ease migrating.
+
+set -e
+TASK="$1"
+shift 1
+
+case "$TASK" in
+  user:create-admin)
```
Comment (ShaneKilkelly) [inline bin/grunt:9]: nice :+1:

**Review by ShaneKilkelly — COMMENTED** [review]
Few comments on genScript:

File: genScript.js (hunk 2 above)
Comment (ShaneKilkelly) [reply genScript.js:4]: Alternatively, could we have the node script call the subprocesses directly? I'm sure there's a subprocess-sync interface in node somewhere, which should make it all relatively easy.

The other alternative would be python, or whatever scripting languages the base image has installed?

Comment (das7pad) [reply genScript.js:4]: As noted in the PR description: we need to fetch multiple details from the `services.js` file -- e.g. get `.repo` and `.name` field for cloning -- which is not straight forward unless we invoke node 100s of times and get one detail at a time (`REPO=$(node ...; NAME=$(node ...); git clone "$REPO" "$NAME"`).

File: genScript.js (hunk 1 above)
Comment (das7pad) [reply genScript.js:9]: Yeah, I do not think this is an issue: Both `services.js` and this script are in the same folder, so write access to one but not the other seems unlikely :wink:

**Review by ShaneKilkelly — APPROVED** [review]

**Comment by das7pad** [comment]
With the web PR for the scripts merged, I've triggered a re-build of the docker image and started it up locally. Logging in/creating new projects/editing docs/compiling/history pane/chat tested.

**Pull request merged**